lib/ql_string.qpp

```diff
@@ -1205,43 +1214,55 @@ string replace(string str, string source, string target, int start = 0, int end
         const char* pattern = t1->c_str();
 
         if (start) {
+            if (start < 0) {
+                start = str->size() + start;
```

Comment:
The `start` and `end` params should be in characters (not bytes). `QoreString::size()` returns size in bytes though.

Reply:
fixed